src/browser/BrowserCbor.cpp

```
@@ -96,8 +96,15 @@ QByteArray BrowserCbor::cborEncodePublicKey(int alg, const QByteArray& first, co
 
         writer.endMap();
     } else if (alg == WebAuthnAlgorithms::EDDSA) {
-        // https://www.rfc-editor.org/rfc/rfc8152#section-13.2
-        writer.startMap(3);
+        writer.startMap(4);
+
+        // Key type
+        writer.append(1);
+        writer.append(getCoseKeyType(alg));
```

I can't comment on the functions directly, but I think the default cases for `getCoseKeyType()` and `getCurveParameter()` should throw errors instead of returning `WebAuthnCoseKeyType::EC2` and  `WebAuthnCurveKey::P256` respectively, as it should be impossible to hit those default cases (as the algorithm is validated before those functions get called), so hitting the default case is a logical error that shouldn't be silently ignored.